src/components/ui4/ButtonUi4.tsx

```diff
@@ -78,16 +78,24 @@ export function ButtonUi4(props: Props) {
     opacity: disabled ? 0.3 : pending ? 0.7 : 1,
     ...sidesToMargin(mapSides(fixSides(marginRem, 0), theme.rem))
   }
+
+  // Show a spinner if waiting on the onPress promise OR if the spinner prop is
+  // manually enabled.
+  const maybeSpinner = !pending && !spinner ? null : <ActivityIndicator color={spinnerColor} style={styles.spinnerCommon} />
+
+  const maybeText =
+    label == null ? null : (
+      <Text adjustsFontSizeToFit minimumFontScale={0.75} numberOfLines={1} style={textStyle}>
+        {label}
+      </Text>
+    )
+
   return (
     <TouchableOpacity disabled={disabled || pending} style={[buttonShadow, styles.borderRadiusCommon]} onPress={handlePress}>
       <LinearGradient {...gradientProps} style={[styles.gradientLayoutCommon, dynamicGradientStyles, styles.borderRadiusCommon]}>
-        {pending ? null : children}
-        {pending || label == null ? null : (
-          <Text adjustsFontSizeToFit minimumFontScale={0.75} numberOfLines={1} style={textStyle}>
-            {label}
-          </Text>
-        )}
-        {!pending && !spinner ? null : <ActivityIndicator color={spinnerColor} style={styles.spinnerCommon} />}
+        {maybeSpinner}
+        {children}
```

Comment:
The current design hides the content when we show the spinner, but you are putting the spinner to the left of the content. This might be fine if we somehow dim the content, but we don't, so I think it will be confusing.

Can we just go back to hiding the content?

```ts
const hideContent = pending || spinning

<LinearGradient ...>
  { hideContent ? null : children }
  { hideContent ? null : maybeText }
  { !hideContent ? null : <ActivityIndicator ...> }
</LinearGradient>
```

Reply:
Wasn't intentional. I was trying to keep it the same but misread the original implementation.